File: 0443-string-compression/0443-string-compression.c
int compress(char* chars, int charsSize) {
    int write = 0;
    int i = 0;
    while(i < charsSize){
        int count = 0;
        int current_char = chars[i];
        while(i < charsSize && current_char == chars[i]){
            count++;
            i++;
        }
        chars[write++] = current_char;
        if(count > 1){
            char digits[5];
            int len = sprintf(digits , "%d" , count);
            for(int j = 0; j < len; j++){
                chars[write++] = digits[j];
            }
        }
    }
    return write;
}

File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        write = 0
        i = 0
        n = len(chars)
        while i < n:
            count = 0
            curr_char = chars[i]
            while i < n and curr_char == chars[i]:
                count += 1
                i += 1
            chars[write] = curr_char
            write += 1
            if count > 1:
                for digits in str(count):
                    chars[write] = digits
                    write += 1
        return write